core/src/main/java/com/rudderstack/android/sdk/core/RudderDataResidencyManager.java

```diff
@@ -47,14 +48,17 @@ class RudderDataResidencyManager {
      *                 .withResidencyServer(ResidencyServer.US)
      *                 .build()
      *     )
-     *      }
+     *     }
      * </pre>
+     *
+     * @return Data Residency URL if present otherwise null
      */
-    void processDataPlaneUrl() {
+    @Nullable
+    String getDataResidencyUrl() {
```

Comment:
Can we improve the function name `getDataResidencyUrl`, 
because currently, we are having two different functions with the same name.

I would suggest renaming the second function at least.